tests/agent_server/deploy/e2e/test_facts_new.py

```diff
@@ -0,0 +1,439 @@
+            "factvalue": None,
+            "requires": [],
+        },
+        {
+            "key": "key6",
+            "value": "value1",
+            "id": "test::Fact[agent1,key=key6],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "skip": False,
+            "skipFact": False,
+            "factvalue": None,
+            "requires": [],
+        },
+        {
+            "key": "key7",
+            "value": "value1",
+            "id": "test::Fact[agent1,key=key7],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "skip": False,
+            "skipFact": False,
+            "factvalue": "",
+            "requires": [],
+        },
+    ]
+
+    resource_states = {
+        "test::Fact[agent1,key=key4]": const.ResourceState.undefined,
+        "test::Fact[agent1,key=key1]": const.ResourceState.undefined,
+        "test::Fact[agent1,key=key5]": const.ResourceState.undefined,
+    }
+
+    async def get_fact(rid, result_code=200, limit=10, lower_limit=2):
+        lower_limit = limit - lower_limit
+        result = await client.get_param(environment, "fact", rid)
+
+        # add minimal nr of reps or failure cases
+        while (result.code != result_code and limit > 0) or limit > lower_limit:
+            limit -= 1
+            await asyncio.sleep(0.1)
+            result = await client.get_param(environment, "fact", rid)
+
+        assert result.code == result_code, result.result
+        return result
+
+    result = await client.put_version(
+        tid=environment,
+        version=version,
+        resources=resources,
+        unknowns=[],
+        version_info={},
+        resource_state=resource_states,
+        compiler_version=get_compiler_version(),
+    )
+    assert result.code == 200
+
+    # THIS IS A BEHAVIOR CHANGE: NO FACTS IF NOT RELEASED
+    # await get_fact("test::Fact[agent1,key=key1]", 503)  # undeployable
+    # await get_fact("test::Fact[agent1,key=key2]")  # normal
+    # await get_fact("test::Fact[agent1,key=key3]", 503)  # not present
+    # await get_fact("test::Fact[agent1,key=key4]", 503)  # unknown
+    # await get_fact("test::Fact[agent1,key=key5]", 503)  # broken
+    # f6 = await get_fact("test::Fact[agent1,key=key6]")  # normal
+    # f7 = await get_fact("test::Fact[agent1,key=key7]")  # normal
+    #
+    # assert f6.result["parameter"]["value"] == "None"
+    # assert f7.result["parameter"]["value"] == ""
+
+    result = await client.release_version(environment, version, True, const.AgentTriggerMethod.push_full_deploy)
+    assert result.code == 200
+
+    await _wait_until_deployment_finishes(client, environment, version)
+
+    await get_fact("test::Fact[agent1,key=key1]", 503, lower_limit=0)  # undeployable
+    await get_fact("test::Fact[agent1,key=key2]")  # normal
+    await get_fact("test::Fact[agent1,key=key3]")  # not present -> present
+    pytest.skip("No unknowns yet!")
```

Comment:
Same comment for the previous skip

Reply:
added to https://github.com/inmanta/inmanta-core/issues/8013